src/test/java/com/google/sps/RootsTest.java

```diff
@@ -0,0 +1,288 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.common.graph.*;
+import com.google.sps.servlets.DataServlet;
+import java.util.HashMap;
+import java.util.HashSet;
+import com.google.sps.data.GraphNode;
+import com.proto.GraphProtos.Node;
+import com.proto.GraphProtos.Node.Builder;
+import com.proto.MutationProtos.Mutation;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class RootsTest {
+  DataServlet servlet;
+
+  // Proto nodes to construct graph with
+  Builder nodeA = Node.newBuilder().setName("A");
+  Builder nodeB = Node.newBuilder().setName("B");
+  Builder nodeC = Node.newBuilder().setName("C");
+
+  GraphNode gNodeA;
```

Comment:
Seems like you don't actually need gNodeA?

Reply:
oh deleted!